cxx4opencl/diff2openclc.txt

```diff
@@ -0,0 +1,120 @@
+=== Difference to OpenCL C
+
+{cpp} for OpenCL targets backwards compatibility to OpenCL C
+in the majority of features. However, it is a different
+language that is derived from {cpp} and therefore it inherits
+its fundamental design principles. Hence {cpp} for OpenCL will
+deviate from OpenCL C in the same areas where {cpp} deviates
+from C.
+
+In this section some of the most common areas where it is
+expected to behave differently are described.
+
+==== Implicit Conversions
+
+{cpp} is much more strict about conversions between types
+especially those that are performed implicitly by the compiler.
+For example it is not possible to convert non-`const` object
+to `const` explicitly. For details please refer to {cpp}17
```

Comment:
I believe the terms in this sentence are swapped.

```suggestion
For example it is not possible to convert a `const` object
to non-`const` implicitly. For details please refer to {cpp}17
```

I'm also suggesting using _implicitly_ as it might be possible to use `const_cast` for this in some specific cases (which is outside the scope of this doc, I think).